azure-pipelines.yml

```diff
@@ -0,0 +1,164 @@
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/release'))
+    jobs:
+     - job: Release
+       pool:
+        vmImage: 'windows-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: PowerShell@2
+          displayName: 'Pre-release versioning - pubspec.yaml'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "(?<=version: ).*" | %{ $_.Matches[0].Value }
+              Write-Host "##vso[task.setvariable variable=currentVersion]$version"
+
+              [string] $versionRegex = '(?<=version: ).*'
+              [string] $pubspecContent = Get-Content -Path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -Raw
+
+              Write-Host " (i) Current version: $version"
+              Write-Host " (i) Original pubspec.yaml content: $pubspecContent"
+
+              [bool] $hasMatches = $pubspecContent -match $versionRegex
+
+              Write-Host " (i) Found matches to replace version? $hasMatches"
+              
+              [string] $newPubspecContent = $pubspecContent -replace $versionRegex, "$version-dev.$(Build.BuildID)"
+
+              Write-Host " (i) Updated pubspec.yaml content: $newPubspecContent"
+
+              $newPubspecContent | Set-Content -Path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml"
+
+        - task: PowerShell@2
+          displayName: 'Pre-release versioning - CHANGELOG.md'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $changelogContent = Get-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md" -Raw
+
+              Write-Host " (i) Current version: $env.currentVersion"
+              Write-Host " (i) Original CHANGELOG.md content: $changelogContent"
+
+              [string] $newChangelogContent = $changelogContent -replace $env:currentVersion, "$env:currentVersion-dev.$(Build.BuildID)"
+
+              Write-Host " (i) Updated CHANGELOG.md content: $newChangelogContent"
+
+              $newChangelogContent | Set-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md"
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+            
+  # ----------- CD Production -----------
+  - stage: ReleaseProd
+    dependsOn: Build
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/master'))
```

Comment:
Should we be releasing from `release` branch instead?

Reply:
long conversation with brett on this one and for the moment this is what we agreed - would be good to get your input on this though as we can evolve this if we're not comfortable with this approach

Reply:
so our decisions were:

- merge to `develop` - trigger CI
- merge to `release` - trigger CI and CD pre-release
- merge to `master` - trigger CI and CD production